pennylane_lightning/core/src/simulators/lightning_tensor/utils/tncuda_utils/tncuda_helpers.hpp

```diff
@@ -194,4 +194,28 @@ inline auto create_swap_wire_pair_queue(const std::vector<std::size_t> &wires)
     return {local_wires, swap_wires_queue};
 }
 
+/**
+ * @brief Check if the provided MPS has the correct dimension for C++
+ * backend.
+ *
+ * @param MPS_shape_dest Dimension list of destination MPS.
+ * @param MPS_shape_source Dimension list of incoming MPS.
+ */
+inline void
+MPSShapeCheck(const std::vector<std::vector<std::size_t>> &MPS_shape_dest,
+              const std::vector<std::vector<std::size_t>> &MPS_shape_source) {
```

Comment:
Could this function not be simplified to something like
```
PL_ABORT_IF_NOT(MPS_shape_dest == MPS_shape_source, "The incoming MPS..." + Pennylane::Util::vector2DToString<std::size_t>(MPS_shape_source) + "... " )

So we don't need to check `MPS_shape_dest == MPS_shape_source` twice, and also don't need to introduce these temporary strings?

Reply:
Nice catch